lib/core/request.c

```diff
@@ -697,6 +697,7 @@ static size_t append_with_limit(char *dst, h2o_iovec_t input, size_t limit)
         memcpy(dst, input.base, input.len);
         return input.len;
     } else {
+        assert(limit >= 3);
```

Comment:
I think we needn't declare this kind of assertions, as we don't put `assert(dst != NULL)`, `assert(input.base != NULL)` and so on. If you were worried about the contract, it would be good enough to declare these contracts in a comment.

Reply:
Okay makes sense. I vaguely remember that I had a similar discussion before. Withdrawing this PR then.